src/transformers/models/vision_encoder_decoder/modeling_tf_vision_encoder_decoder.py

```diff
@@ -0,0 +1,715 @@
+            >>> decoder_tokenizer = AutoTokenizer.from_pretrained("gpt2")
+
+            >>> # initialize a bert2gpt2 from a pretrained BERT and GPT2 models. Note that the cross-attention layers will be randomly initialized
+            >>> model = TFVisionEncoderDecoderModel.from_encoder_decoder_pretrained(
+            ...     'google/vit-base-patch16-224-in21k', 'gpt2'
+            ... )
+
+            >>> url = "http://images.cocodataset.org/val2017/000000039769.jpg"
+            >>> img = Image.open(requests.get(url, stream=True).raw)
+
+            >>> # forward
+            >>> pixel_values = feature_extractor(images=img, return_tensors="tf").pixel_values  # Batch size 1
+            >>> decoder_input_ids = decoder_tokenizer("Linda Davis", return_tensors="tf").input_ids  # Batch size 1
+            >>> outputs = model(pixel_values=pixel_values, decoder_input_ids=decoder_input_ids)
+
+            >>> # training
+            >>> outputs = model(pixel_values=pixel_values, decoder_input_ids=decoder_input_ids, labels=decoder_input_ids)
+            >>> loss, logits = outputs.loss, outputs.logits
+
+            >>> # save and load from pretrained
+            >>> model.save_pretrained("vit-gpt2")
+            >>> model = TFVisionEncoderDecoderModel.from_pretrained("vit-gpt2")
+
+            >>> # generation
+            >>> generated = model.generate(pixel_values, decoder_start_token_id=model.config.decoder.bos_token_id)
+
+        """
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        kwargs_encoder = {argument: value for argument, value in kwargs.items() if not argument.startswith("decoder_")}
+
+        kwargs_decoder = {
+            argument[len("decoder_") :]: value for argument, value in kwargs.items() if argument.startswith("decoder_")
+        }
+
+        # Let the user be responsible for the expected format.
+        if encoder_outputs is not None:
+            if return_dict and not isinstance(encoder_outputs, ModelOutput):
+                raise ValueError(
+                    "If `return_dict=True` and `encoder_outputs` is provided, it should be an instance of "
+                    f"`ModelOutput`. Got an instance {type(encoder_outputs)} for `encoder_outputs`."
+                )
+
+        if encoder_outputs is None:
+
+            encoder_processing_inputs = {
+                "func": self.encoder.call,
+                "config": self.encoder.config,
+                "input_ids": pixel_values,
+                "output_attentions": output_attentions,
+                "output_hidden_states": output_hidden_states,
+                "return_dict": return_dict,
+                "training": training,
+                "kwargs_call": kwargs_encoder,
+            }
+
+            # Add arguments to encoder from `kwargs_encoder`
+            for k, v in kwargs_encoder.items():
+                encoder_processing_inputs[k] = v
```

Comment:
There is a function for this:
```suggestion
            encoder_processing_inputs.update(kwargs_encoder)
```